diff --git a/source/rofi.c b/source/rofi.c
index 37ad16f0..58fca78a 100644
--- a/source/rofi.c
+++ b/source/rofi.c
@@ -76,6 +76,19 @@
 
 #endif
 
+/** Terrible hack */
+static void terrible_hack_for_appimage(void)
+{
+  if ( g_getenv("APPDIR") ) {
+    g_unsetenv("APPDIR");
+    g_unsetenv("GIO_MODULE_DIR");
+    g_unsetenv("GSETTINGS_SCHEMA_DIR");
+    g_unsetenv("GDK_PIXBUF_MODULEDIR");
+    g_unsetenv("GDK_PIXBUF_MODULE_FILE");
+    g_unsetenv("GTK_PATH");
+  }
+}
+
 /** Location of pidfile for this instance. */
 char *pidfile = NULL;
 /** Location of Cache directory. */
@@ -1299,6 +1312,8 @@ int main(int argc, char *argv[]) {
   rofi_icon_fetcher_init();
   TICK_N("Icon fetcher initialize");
 
+  terrible_hack_for_appimage();
+
   gboolean kill_running = FALSE;
   if (find_arg("-replace") >= 0) {
     kill_running = TRUE;
